server.go

```diff
@@ -260,6 +266,18 @@ func NewServer(opts ...ServerOption) (*Server, error) {
 	return s, nil
 }
 
+func (s *Server) expandDataDirName() error {
```

Comment:
It seems like making this method specific to `s.Holder.Path` is not ideal.
What do you think about making it more generic (so that it simply expands a `dir` that is provided).
Then, for example, it could be applied to the `dir` argument supplied to `OptServerDataDir()`. And that way it would be correct for both `s.Holder.Path` and `s.Cluster.Path`. Does that seem right to you?

Reply:
That makes sense. Updated the PR.